great_expectations/agent/agent.py

```diff
@@ -0,0 +1,122 @@
+import os
+from typing import Dict, Optional
+
+from pydantic.dataclasses import dataclass
+from pydantic import AmqpDsn
+from great_expectations import get_context
+from great_expectations.agent.event_handler import ShutdownRequest, EventHandler
+from great_expectations.agent.message_service.rabbit_mq_client import (
+    RabbitMQClient,
+    ClientError,
+)
+from great_expectations.agent.message_service.subscriber import (
+    Subscriber,
+    OnMessageCallback,
+    SubscriberError,
+)
+from great_expectations.agent.models import Event
+from great_expectations.data_context import CloudDataContext
+
+
+HandlerMap = Dict[str, OnMessageCallback]
+
+
+@dataclass(frozen=True)
+class GXAgentConfig:
+    """GXAgent configuration.
+    Attributes:
+        organization_id: GX Cloud organization identifier
+        broker_url: address of broker service
+    """
+
+    organization_id: str
+    broker_url: AmqpDsn
+
+
+class GXAgent:
+    """
+    Run GX in any environment from GX Cloud.
+
+    To start the agent, install Python and great_expectations and run `gx-agent`.
+    The agent loads a DataContext configuration from GX Cloud, and listens for
+    user events triggered from the UI.
+    """
+
+    def __init__(self):
+        print("Initializing GX-Agent")
```

Comment:
`non-blocking`
As a followup, I'd love to see us use a logger instead of printing for all these messages and make the level configuration when running the agent.

Reply:
agree, i had explored that as an initial implementation, but was much more straightforward to just go with print (also, is inline with the gx cli). let's consider it as followup work.